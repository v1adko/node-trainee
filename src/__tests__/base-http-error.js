import BaseHttpError from '../lib/errors/base-http-error';

const throwAndCatchError = (error) => {
  try {
    throw error;
  } catch (catchedError) {
    return catchedError;
  }
};

describe('Test base class for all Http errors BaseHttpErrors', () => {
  it('should check inherit from ERROR', () => {
    const error = new BaseHttpError();

    expect(error instanceof Error).toBe(true);
    expect(error instanceof BaseHttpError).toBe(true);
  });

  it('should check standart behavior of error', () => {
    const baseError = new BaseHttpError();
    const error = throwAndCatchError(baseError);

    expect(error).toBe(baseError);
  });
});

class TestError extends BaseHttpError {
  constructor() {
    super('test message.', 500);
  }

  getResponseObject = () =>
    super.getResponseObject({ testResposeField: 'test text of field' });
}

describe('Test inherit for BaseHttpErrors', () => {
  it('should check inherit from BaseHttpErrors', () => {
    const error = new TestError();

    expect(error instanceof Error).toBe(true);
    expect(error instanceof BaseHttpError).toBe(true);
    expect(error instanceof TestError).toBe(true);
  });

  it('should check standart behavior of error', () => {
    const baseError = new TestError();
    const error = throwAndCatchError(baseError);

    expect(error).toBe(baseError);
    expect(error instanceof Error).toBe(true);
    expect(error instanceof BaseHttpError).toBe(true);
    expect(error instanceof TestError).toBe(true);
  });

  it('should return response object', () => {
    const baseError = new TestError();
    const responseObject = baseError.getResponseObject();

    expect(responseObject).toMatchSnapshot();
  });

  it('should check adding of messages', () => {
    const baseError = new TestError();
    const message1 = 'First dditiuonal test message of error.';
    const message2 = 'Second dditiuonal test message of error.';
    baseError.addMessage(message1);
    baseError.addMessage(message2);

    expect(baseError).toMatchSnapshot();
  });
});
