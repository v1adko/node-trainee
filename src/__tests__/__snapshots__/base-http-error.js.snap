// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test inherit for BaseHttpErrors should check adding of messages 1`] = `[Error: TestError(500): test message. First dditiuonal test message of error. Second dditiuonal test message of error.]`;

exports[`Test inherit for BaseHttpErrors should return response object 1`] = `
Object {
  "code": 500,
  "message": "TestError(500): test message.",
  "messages": Array [
    "TestError(500): test message.",
  ],
  "name": "TestError",
  "testResposeField": "test text of field",
}
`;
